app/models/reports/instrument_unavailable_report.rb

```diff
@@ -0,0 +1,73 @@
+module Reports
+
+  class InstrumentUnavailableReport
+
+    QUANTITY_INDEX = 3
+    HOURS_INDEX = 4
+
+    attr_reader :facility, :start_time, :end_time
+
+    include FacilityReservationsHelper
+
+    def initialize(facility, date_start, date_end)
+      @facility = facility
+      @start_time = date_start.beginning_of_day
+      @end_time = date_end.end_of_day
+    end
+
+    def rows
+      @rows ||= raw_values.map do |row|
+        row[:hours] = format("%.2f", row.delete(:seconds) / 3600.0).to_f
+        row
+      end.map(&:values)
+    end
+
+    def total_hours
+      rows.map { |row| row[HOURS_INDEX] }.sum
+    end
+
+    def total_quantity
+      rows.map { |row| row[QUANTITY_INDEX] }.sum
+    end
+
+    private
+
+    def raw_values
+      query.each_with_object({}) do |row, report|
```

Comment:
This report seems like it might be more complicated than it needs to be.

Using `group_by` and `sum` might simplify things a bit:

```ruby
grouped = query.group_by { |res| [res.product.name, res.type, res.category] }
rows = grouped.each do |key, reservations|
   key + [reservations.count, format_seconds(reservations.sum_by { |r| duration_in_sections(r) })]
end
```